### DISCUSSION TITLE
Error:  compiling  example List.ino esp32

### DISCUSSION THREAD

**Discussion by maansch** [forntoh/LcdMenu#99] [Q&A] [answered]
Hi Thomas,
try to use the List.ino, during compiling I receive these error msg.

/home/mas/A_Prog_A/Arduino/libraries/LcdMenu/src/ItemList.h: In constructor 'ItemList::ItemList(const char*, String*, uint8_t, fptrInt)':
/home/mas/A_Prog_A/Arduino/libraries/LcdMenu/src/ItemList.h:34:21: error: 'ItemList::items' will be initialized after [-Werror=reorder]
     String* items = NULL;     ///< Pointer to an array of items
                     ^
/home/mas/A_Prog_A/Arduino/libraries/LcdMenu/src/ItemList.h:33:24: error:   'void (* ItemList::callback)(uint8_t)' [-Werror=reorder]
     fptrInt callback = NULL;  ///< Pointer to a callback function
                        ^
In file included from /tmp/arduino_modified_sketch_881776/List.ino:21:0:
/home/mas/A_Prog_A/Arduino/libraries/LcdMenu/src/ItemList.h:47:5: error:   when initialized here [-Werror=reorder]
     ItemList(const char* key, String* items, const uint8_t itemCount,
     ^
cc1plus: some warnings being treated as errors

Bibliothek LcdMenu in Version 3.0.1 im Ordner: /home/mas/A_Prog_A/Arduino/libraries/LcdMenu  wird verwendet
Bibliothek LiquidCrystal_I2C in Version 1.1.4 im Ordner: /home/mas/A_Prog_A/Arduino/libraries/LiquidCrystal_I2C  wird verwendet
Bibliothek Wire in Version 1.0.1 im Ordner: /home/mas/arduinoide/arduino-1.8.19_ESP32/portable/packages/esp32/hardware/esp32/1.0.6/libraries/Wire  wird verwendet

exit status 1
Fehler beim Kompilieren für das Board ESP32 Dev Module.

Do you have an idea where the error could be?
I would be grateful for your help.

br
Marc-Andre

Code: example List.ino
environment : Arduino: 1.8.19 (Linux),
Board Typ: esp32
AZ-Delivery ESP-32Development-Board, ESP-WROOM-32 Prozessor 160MHz, Tensilica L108 32 bit Dual-Core CPU,  512 KB SRAM & 16 MB Flashspeicher.
Bord driver for testing:
Core Version 2.9.0 
Core Version 1.6.0

**Comment by forntoh** [comment]
Is it specific to this board? Have you tried other boards?

Also, it will be best if you convert this discussion into an issue.

**Comment by maansch** [comment]
Hi,
I have only these type of esp32 board
Not tested with other stuff.
br
Marc-Andre

**Reply by forntoh** [reply]
Try v3.1.0, there are multiple changes

**Comment by maansch** [comment]
did not work with v3.1.0 
error msg:
In file included from /tmp/.arduinoIDE-unsaved2023516-724100-uv7vra.fnw5/List/List.ino:21:0:
... /libraries/LcdMenu/src/ItemList.h: In constructor 'ItemList::ItemList(const char*, String*, uint8_t, fptrInt)':

... /libraries/LcdMenu/src/ItemList.h:35:19: error: 'ItemList::itemCount' will be initialized after [-Werror=reorder]
     const uint8_t itemCount;  ///< The total number of items in the list
                   ^
... /sketches/DAE3657AB8AC31D158E005E1875B5760/sketch/List.ino.cpp:1:
... /libraries/LcdMenu/src/ItemList.h:32:24: error:   'void (* ItemList::callback)(uint8_t)' [-Werror=reorder]
     fptrInt callback = NULL;  ///< Pointer to a callback function
                        ^
In file included from /tmp/.arduinoIDE-unsaved2023516-724100-uv7vra.fnw5/List/List.ino:21:0:
... /libraries/LcdMenu/src/ItemList.h:47:5: error:   when initialized here [-Werror=reorder]
     ItemList(const char* key, String* items, const uint8_t itemCount,
     ^
cc1plus: some warnings being treated as errors                   
exit status 1
Compilation error: exit status 1

-----------
previous error message v3.0.1
... /libraries/LcdMenu/src/ItemList.h: In constructor 'ItemList::ItemList(const char*, String*, uint8_t, fptrInt)':

... /libraries/LcdMenu/src/ItemList.h:34:21: error: 'ItemList::items' will be initialized after [-Werror=reorder]
String* items = NULL; ///< Pointer to an array of items
^

... /libraries/LcdMenu/src/ItemList.h:33:24: error: 'void (* ItemList::callback)(uint8_t)' [-Werror=reorder]
fptrInt callback = NULL; ///< Pointer to a callback function
^

In file included from /tmp/arduino_modified_sketch_881776/List.ino:21:0:
... /libraries/LcdMenu/src/ItemList.h:47:5: error: when initialized here [-Werror=reorder]
ItemList(const char* key, String* items, const uint8_t itemCount,
^

**Comment by forntoh** [comment]
It should be fixed in v3.1.1

**Comment by maansch** [comment]
Hi Thomas,
compiling without error msg work’s now. 

but it’s not possible to use the list.

Scrolling by UP / DOWN works.

ENTER LEFT/RIGHT has no function.

Br
Marc-Andre

**Reply by forntoh** [reply]
Created a new issue for that #105, thanks for reporting

**Reply by maansch** [reply, marked as answer]
kein Ding :+1:

**Comment by B1ackt34** [comment]
Does this library works with ESP32 dev module?
